Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -351,10 +351,14 @@ extension ChannelDTO {
         // We can't pass bools directly to NSPredicate so we have to use integers
         let isHidden = NSPredicate(format: "isHidden == %i", query.filter.hiddenFilterValue == true ? 1 : 0)
 
-        let subpredicates = [
+        var subpredicates: [NSPredicate] = [
             matchingQuery, notDeleted, isHidden
         ]
 
+        if let filterPredicate = query.filter.predicate {
```

Comment:
What's the best way to allow that be configured and in case it's causing issues, customers can disable it?